python.d/tor.chart.py

```diff
@@ -0,0 +1,109 @@
+#
+# Create a configuration file at /etc/netdata/tor.conf
+#
+# local:
+#     connect_method: 'port'
+#     password: 'REPLACEME'
+#     port: 9051
+#     socket_path: '/tmp/REPLACEME'
+#
+# You can use "port" or "socket" as "connect_method"
+# The "port" and "socket_path" params will be used accodingly
+
+from bases.FrameworkServices.SocketService import SocketService
+
+try:
+    import stem
+    import stem.connection
+    import stem.control
+    STEM_AVAILABLE = True
+except ImportError:
+    STEM_AVAILABLE = False
+
+NAME = "Tor"
+
+# default module values
+# update_every = 10
+priority = 90000
+retries = 60
+
+ORDER = ['traffic']
+CHARTS = {
+    'traffic': {
+        'options': [None, "Tor traffic", 'bytes/s', 'Tor traffic',
+                    'tor.traffic', 'line'],
+        'lines': [
+            ["read", "read", "incremental"],
+            ["write", "write", "incremental"],
+        ]
+    }
+}
+
+
+class Service(SocketService):
+    """Provide netdata service for Tor"""
+    def __init__(self, configuration=None, name=None):
+        SocketService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+
+    def _authenticate(self):
+        connect_method = self.configuration.get('connect_method', 'port')
+        password = self.configuration.get('password', None)
+        if connect_method == 'port':
+            port = self.configuration.get('port', 9051)
+            self._controller = stem.control.Controller.from_port(port=port)
+        else:
+            socket_path = self.configuration.get('socket_path', '')
+            self._controller = stem.control.Controller.from_socket_file(
+                path=socket_path)
+
+        try:
+            self._controller.authenticate(password=password)
+        except stem.connection.AuthenticationFailure as e:
+            self.error('Authentication failed ({})'.format(e))
+
+    def check(self):
+        """
+        Parse configuration, check if Tor is available
+        :return: boolean
+        """
+        self._parse_config()
+        if not STEM_AVAILABLE:
+            self.error("The stem library is missing")
+            return False
+
+        try:
+            self._authenticate()
+            return True
+        except Exception as e:
+            self.error(str(e))
+            return False
+
+    def get_metric(self, name):
+        try:
+            return self._controller.get_info(name)
+        except Exception:
+            # Try to recreate the controller and authenticate - just once
+            self._authenticate()
+            return self._controller.get_info(name)
+
+    def _get_data(self):
+        """Get host metrics
+        :return: dict
+        """
+        try:
+            tr = self.get_metric('traffic/read')
+            tw = self.get_metric('traffic/written')
+            data = dict(
+                read=int(tr),
+                write=int(tw)
+            )
```

Comment:
Please update the above to the standard dictionary literal syntax.  It should end up being something like this:

```
data = {
    'read': int(tr),
    'write': int(tw),
}
```